src/doc/unstable-book/src/compiler-flags/extern options.md

```diff
@@ -0,0 +1,19 @@
+The behavior of the `--extern` flag can be modified with `noprelude`, `priv` or `nounused` options.
+
+This is unstable feature, so you have to provide `-Zunstable-options` to get this feature working.
+
+## Examples
+
+Use your own build of the `core` crate.
+
+`rustc main.rs -Z unstable-options --extern noprelude:core=libcore.rlib`
+
+To use multiple options, separate them with a comma:
+
+`rustc main.rs -Z unstable-options --extern noprelude,priv,nounused:mydep=mydep.rlib`
+
+## Options
+
+* `noprelude`: Do not add the crate to the external prelude. If used, it will need to be imported using `extern crate`.
+* `priv`: Mark the crate as a private dependency for the `exported_private_dependencies` lint.
```

Comment:
can we link to https://doc.rust-lang.org/stable/rustc/lints/listing/warn-by-default.html#exported-private-dependencies here?